src/Symfony/Component/Validator/Constraints/AllValidator.php

```diff
@@ -44,12 +44,12 @@ public function validate($value, Constraint $constraint)
             $validator = $context->getValidator()->inContext($context);
 
             foreach ($value as $key => $element) {
-                $validator->atPath('['.$key.']')->validate($element, $constraint->constraints);
+                $validator->atPath('['.$key.']')->validate($element, $constraint->constraints, $constraint->groups);
```

Comment:
Shouldn't this be `$context->getGroup()` too?

Reply:
To give more context: This implementation suffers the same implications as discussed in https://github.com/symfony/symfony/pull/23480#discussion_r153483114.

Reply:
The `Composite` constraint is special, `All` and `Collection` inherit it.
The rules are explained here https://github.com/symfony/symfony/blob/master/src/Symfony/Component/Validator/Constraints/Composite.php#L34-L48, we need to not break them.
So I tend to agree with @xabbuh that we should not just rely on the `groups` public property of nested constraints to resolve any issue.